parser_test.go

```diff
@@ -3239,6 +3239,7 @@ func (s *testParserSuite) TestTablePartition(c *C) {
 	table := []testCase{
 		{"ALTER TABLE t1 TRUNCATE PARTITION p0", true, "ALTER TABLE `t1` TRUNCATE PARTITION `p0`"},
 		{"ALTER TABLE t1 TRUNCATE PARTITION p0, p1", true, "ALTER TABLE `t1` TRUNCATE PARTITION `p0`,`p1`"},
+		{"ALTER TABLE t1 TRUNCATE PARTITION ALL", true, "ALTER TABLE `t1` TRUNCATE PARTITION ALL"},
```

Comment:
Could you add a test case showing that both

```sql
ALTER TABLE t1 TRUNCATE PARTITION ALL, p0;
ALTER TABLE t1 TRUNCATE PARTITION p0, ALL;
```

will be *parse failure*? Thanks.

Reply:
Thanks for your review. But according to the syntax definition and MySQL instance on my machine, both case shouldn't work.
```
➜  ~ mysql -uroot -h127.0.0.1 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 11
Server version: 8.0.17 MySQL Community Server - GPL

Copyright (c) 2000, 2018, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> alter table demo.t1 truncate partition p1;
ERROR 1505 (HY000): Partition management on a not partitioned table is not possible
mysql> alter table demo.t1 truncate partition all;
ERROR 1505 (HY000): Partition management on a not partitioned table is not possible
mysql> alter table demo.t1 truncate partition p1, all;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'all' at line 1
mysql> alter table demo.t1 truncate partition all, p1;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ', p1' at line 1
mysql> show variables like "%version%";
+--------------------------+------------------------------+
| Variable_name            | Value                        |
+--------------------------+------------------------------+
| immediate_server_version | 999999                       |
| innodb_version           | 8.0.17                       |
| original_server_version  | 999999                       |
| protocol_version         | 10                           |
| slave_type_conversions   |                              |
| tls_version              | TLSv1,TLSv1.1,TLSv1.2        |
| version                  | 8.0.17                       |
| version_comment          | MySQL Community Server - GPL |
| version_compile_machine  | x86_64                       |
| version_compile_os       | Linux                        |
| version_compile_zlib     | 1.2.11                       |
+--------------------------+------------------------------+
11 rows in set (0.00 sec)
```

Reply:
Thanks for reminding, I've added these case.